### ISSUE TITLE
Hono Context / Cookie cached ? and compress plugin problem

### ISSUE THREAD

**Issue by fortezhuo** [vikejs/vike-node#7]
Hi,

I played with Hono and Vike Node, and found 2 issues

- while using cookies
- compress plugin cause page rendered not correctly

How to reproduce : 
1. Clone below repo / copy below script
2. Run http://localhost:3000/redirect to refresh cookie
3. Page will be redirect to / and cookie will be displayed
4. Run step no. 2 and get cookie value on page will display prev value

Source : https://github.com/fortezhuo/vike-node-hono

app.js
```
const app = new Hono()
app.post('_telefunc', async (ctx) => {
  const { url, method } = ctx.req
  const body = await ctx.req.text()
  const httpResponse = await telefunc({ url, method, body })
  return new Response(httpResponse.body, {
    status: httpResponse.statusCode
  })
})
app.get("/redirect",async (c)=>{
 setCookie(c,"cookie",Math.random())
  return c.redirect("http://localhost:3000")
})
app.get("/cookie",async (c)=>{
 const cookie = getCookie(c,"cookie")
   return c.json({cookie})
 })
 
app.use(vike({
  pageContext(c){
    const cookie = getCookie(c,"cookie")
     return {
      cookie
    }
  },
  static:{
    root:"./dist/client"
  }
}))
```

config.js
```
const config = {
  passToClient: ["cookie"],
  // https://vike.dev/extends
  extends: vikeReact,
  prerender: false
}
```

+Page.jsx

```

function Page() {
  const { cookie } = usePageContext()
  return (
    <>
      <h1>Welcome</h1>
      This page is: &&&
      <ul>
        <li>Pre-rendered</li>
        <li>Static html generated</li>
        <li>Interactive</li>
      </ul>
      cookie : {cookie}
      <Counter />
    </>
  )
}
```

Screenshot : 
Vike Page has cookie cached 
<img width="758" alt="image" src="https://github.com/user-attachments/assets/20f7de1e-6b7f-4523-aedf-a4c0c6e74108">

Hono Page get cookie correctly
<img width="758" alt="image" src="https://github.com/user-attachments/assets/996b3fb0-3cef-407e-84f3-8e6e5cff411d">

5. Build the repo with compress plugin enabled

<img width="758" alt="image" src="https://github.com/user-attachments/assets/6e9388ef-c033-4045-b5e1-cad0059a61a5">

Updated : 
I have reproduce using batijs and https://github.com/phonzammi/vike-hono-example 's template, seems compress plugin problem related to vike-react https://github.com/vikejs/vike-react/issues/143

**Comment by brillout** [comment]
Regarding the compression issue, try `0.4.191-commit-d2ca712`. Does it solve the issue?

**Comment by fortezhuo** [comment]
> Regarding the compression issue, try `vike@0.4.191-commit-d2ca712`. Does it solve the issue?

I have tried commit-d2ca712, the issue still persist

**Comment by nitedani** [comment]
This is an issue in the `hono/compress` package: https://github.com/honojs/hono/blob/main/src/middleware/compress/index.ts

It compresses everything regardless of what it is. Even if the response is already compressed, uncompressible, etc.. it compresses it again. `vike-node` already handles compression for the html, and `hono/compress` compresses it again without checking anything.

**Comment by nitedani** [comment]
The stale context issue is fixed in `vike-node@0.1.16`

**Comment by nitedani** [comment]
https://github.com/honojs/hono/pull/3317

**Comment by fortezhuo** [comment]
> The stale context issue is fixed in `vike-node@0.1.16`

Confirmed worked now.. with 0.1.16 . I will close this issue.. thanks for your quick response and solution

**Comment by brillout** [comment]
> double compression

I wonder if there is an easy way to detect this and then show a `assertWarning()`/`assertUsage()`. Could be worth it if it's easy to implement.
